perf_test/KokkosKernels_perf_test_utilities.hpp

```diff
@@ -0,0 +1,78 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+//
+// Created by Berger-Vergiat, Luc on 2/6/23.
+//
+
+#ifndef KOKKOSKERNELS_PERF_TEST_UTILITIES_HPP
+#define KOKKOSKERNELS_PERF_TEST_UTILITIES_HPP
+
+// Namepsace that defines common utilities
+// for performance tests
+namespace perf_test {
+
+  struct CommonInputParams {
+    int use_cuda     = 0;
+    int use_hip      = 0;
+    int use_sycl     = 0;
+    int use_openmp   = 0;
+    int use_threads  = 0;
+  };
+
+  std::string list_common_options() {
+
+    std::ostringstream common_options;
+    common_options << "\t[Required] BACKEND:\n"
+		   << "\t\t'--threads [numThreads]' |\n"
+		   << "\t\t'--openmp [numThreads]' |\n"
+		   << "\t\t'--cuda [deviceIndex]' |\n"
+		   << "\t\t'--hip [deviceIndex]' |\n"
+		   << "\t\t'--sycl [deviceIndex]'\n\n"
+		   << "\tIf no parallel backend is requested, Serial will be used "
+      "(if enabled)\n\n";
+
+    return common_options.str();
+  }
+
+  std::vector<int> parse_common_options(CommonInputParams& params, int argc, char** argv) {
+    // Keep track of validated intput params
+    std::vector<int> validated_input(argc, 0);
+
+    for (int argIdx = 1; argIdx < argc; ++argIdx) {
+      if (0 == Test::string_compare_no_case(argv[argIdx], "--threads")) {
+	params.use_threads = atoi(argv[++argIdx]);
+	validated_input[argIdx] = 1;
+      } else if (0 == Test::string_compare_no_case(argv[argIdx], "--openmp")) {
+	params.use_openmp = atoi(argv[++argIdx]);
+	validated_input[argIdx] = 1;
+      } else if (0 == Test::string_compare_no_case(argv[argIdx], "--cuda")) {
+	params.use_cuda = atoi(argv[++argIdx]) + 1;
+	validated_input[argIdx] = 1;
+      } else if (0 == Test::string_compare_no_case(argv[argIdx], "--hip")) {
+	params.use_hip = atoi(argv[++argIdx]) + 1;
+	validated_input[argIdx] = 1;
+      } else if (0 == Test::string_compare_no_case(argv[argIdx], "--sycl")) {
+	params.use_sycl = atoi(argv[++argIdx]) + 1;
+	validated_input[argIdx] = 1;
+      }
```

Comment:
How about instead of returning a flag for each arg index whether it's a common param, just return a vector of strings that has all the remaining args? That would make it easier for each perf test to parse and validate its own specific options.

This also needs bounds checking before reading ``argv[++argIdx]`` - for example if ``--cuda`` is the last argument, this will read past the end of the argv array.

Reply:
And yeah, I missed the bounds checking issue in spadd again, but here we can fix it once and for all :)